ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1718,6 +1722,34 @@ public Table apply(org.apache.hadoop.hive.metastore.api.Table table) {
     }
   }
 
+  /**
+   * Validate if given materialized view has SELECT privileges for current user
+   * @param cachedMVTable
+   * @return false if user does not have privilege otherwise true
+   * @throws HiveException
+   */
+  private boolean checkPrivillegeForMV(final Table cachedMVTable) throws HiveException{
```

Comment:
Can we move this to HiveMaterializedViewUtils?

There is also a typo: `Privillege`